### DISCUSSION TITLE
Apps included in AppStore fluctuate day to day

### DISCUSSION THREAD

**Discussion by robin-a-meade** [jbangdev/jbang#1784] [Q&A] [answered]
I noticed that the apps included in the AppStore seem to fluctuate day to day.

Consider, for example, `echo@quintesse`:

It was removed Mar 26 in commit
https://github.com/jbangdev/jbang.dev/commit/1ab60621b5f17e43e4288a44e9658a99e155f9bf

It was added Mar 27 in commit
https://github.com/jbangdev/jbang.dev/commit/2a4d564a3cfe4c23207195c7d199fcd8c947a678

It was removed Apr 1 in commit
https://github.com/jbangdev/jbang.dev/commit/3c28557cee7bcb7c1fc42be1026b08419e2c65e8

Is this because of github rate limiting the search for `jbang-catalog.json` files?

**Comment by maxandersen** [comment, marked as answer]
Yes :( 

I've not found a way that avoids being rate limited when running in GitHub actions. 

Running it locally with the same key I don't hit the limit. 

Nite, it's the actual search we get rate limited on afaics not the fetch about each file. 

If anyone has ideas how to avoid hitting it that would be awesome. 

I've also considered making the list less dynamic and just keep things forever so the list won't fluctuate this much.

**Reply by quintesse** [reply]
> If anyone has ideas how to avoid hitting it that would be awesome.

Not sure, but it seems jobs are actually allowed to run up to 6 hours, perhaps just do the queries with big delays between them?

**Reply by quintesse** [reply]
Another idea would be to use the search only to _add_ things, but use the catalog file fetch to detect if something was removed. Still, theoretically this might mean that some catalogs might never get added if the rate limit kicks in before they are ever found (something that might happen more often the more catalogs that are created).
